mlrun/serving/server.py

```diff
@@ -547,14 +541,41 @@ async def async_execute_graph(
 ) -> list[Any]:
     spec = mlrun.utils.get_serving_spec()
 
-    source_filename = spec.get("filename", None)
     namespace = {}
-    if source_filename:
-        with open(source_filename) as f:
-            exec(f.read(), namespace)
+    code = os.getenv("MLRUN_EXEC_CODE")
+    if code:
+        code = base64.b64decode(code).decode("utf-8")
+        exec(code, namespace)
+    else:
+        source_filename = spec.get("filename", None)
+        if source_filename:
+            with open(source_filename) as f:
+                exec(f.read(), namespace)
```

Comment:
Why is this needed? IIUC this is to be in line with existing options in a kubejob, but what is the situation where you want to run a serving graph as job and also provide code to execute? Why allow this?